urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/INDISexagesimalFormatter.java

```diff
@@ -0,0 +1,355 @@
+
+  /**
+   * Constructs an instance of
+   * <code>INDISexagesimalFormatter</code> with a particular format.
+   *
+   * @param format The desired format
+   * @throws IllegalArgumentException if the format is not correct: begins with
+   * %, ends with m and specifies a length and fractionLength in the form
+   * length.fractionLength. Valid fractionLengths are 3, 5, 6, 8 and 9. For
+   * example %5.3m.
+   */
+  public INDISexagesimalFormatter(String format) throws IllegalArgumentException {
+    this.format = format;
+
+    checkFormat();
+  }
+
+  /**
+   * Gets the format of this formatter.
+   *
+   * @return the format of this formatter.
+   */
+  public String getFormat() {
+    return format;
+  }
+
+  /**
+   * Checks the specified format string.
+   *
+   * @throws IllegalArgumentException if the format string is not valid: begins
+   * with %, ends with m and specifies a length and fractionLength in the form
+   * length.fractionLength. Valid fractionLengths are 3, 5, 6, 8 and 9. For
+   * example %5.3m.
+   */
+  private void checkFormat() throws IllegalArgumentException {
+    if (!format.startsWith("%")) {
+      throw new IllegalArgumentException("Number format not starting with %");
+    }
+
+    if (!format.endsWith("m")) {
+      throw new IllegalArgumentException("Sexagesimal format not recognized (not ending m)");
+    }
+
+    String remaining = format.substring(1, format.length() - 1);
+
+    int dotPos = remaining.indexOf(".");
+
+    if (dotPos == -1) {
+      throw new IllegalArgumentException("Sexagesimal format not correct (no dot)");
+    }
+
+    String l = remaining.substring(0, dotPos);
+    String frLength = remaining.substring(dotPos + 1);
+
+    try {
+      length = Integer.parseInt(l);
+      fractionLength = Integer.parseInt(frLength);
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Illegal sexagesimal length or fraction length");
+    }
+
+    if ((fractionLength != 3) && (fractionLength != 5) && (fractionLength != 6) && (fractionLength != 8) && (fractionLength != 9)) {
+      throw new IllegalArgumentException("Illegal sexagesimal fraction length");
+    }
+  }
+
+  /**
+   * Parses a sexagesimal number. DO NOT USE IT. THIS IS A PRELIMINARY VERSION
+   * AND DOES NOT WORK AS EXPECTED. THIS METHOD WILL DISAPEAR IN FUTURE VERSIONS
+   * OF THE CLASS.
+   *
+   * @param number NOT USED
+   * @return NOT USED
+   * @throws IllegalArgumentException
+   * @deprecated
+   */
+  @Deprecated
+  public double parseSexagesimal2(String number) throws IllegalArgumentException {
+    number = number.replace(' ', ':');
+    number = number.replace(';', ':');
+
+//    System.out.println(" ->" + number + " ; " + format);
+//    System.out.flush();
+
+    if (number.indexOf(":") == -1) {  // If there are no separators maybe they have sent just a single double
+      try {
+        double n = Double.parseDouble(number);
+
+        return n;
+      } catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 157, 159.